cmd/launcher/flare.go

```diff
@@ -0,0 +1,331 @@
+		insecureGRPC    = env.Bool("KOLIDE_LAUNCHER_INSECURE_GRPC", false)
+		flareSocketPath = env.String("FLARE_SOCKET_PATH", filepath.Join(os.TempDir(), "flare.sock"))
+
+		certPins [][]byte
+		rootPool *x509.CertPool
+	)
+	flagset.Usage = commandUsage(flagset, "launcher flare")
+	if err := flagset.Parse(args); err != nil {
+		return err
+	}
+
+	id := ulid.New()
+	b := new(bytes.Buffer)
+	reportName := fmt.Sprintf("kolide_launcher_flare_report_%s", id)
+	tarOut, err := os.Create(fmt.Sprintf("%s.tar.gz", reportName))
+	if err != nil {
+		fatal(b, err)
+	}
+	defer func() {
+		if err := tarOut.Close(); err != nil {
+			fatal(b, err)
+		}
+	}()
+
+	tw := tar.NewWriter(tarOut)
+
+	// create directory at root of tar file
+	baseDir := filepath.ToSlash(reportName)
+	hdr := &tar.Header{
+		Name:     baseDir + "/",
+		Mode:     0755,
+		ModTime:  time.Now().UTC(),
+		Typeflag: tar.TypeDir,
+	}
+
+	if err := tw.WriteHeader(hdr); err != nil {
+		fatal(b, err)
+	}
+
+	defer func() {
+		hdr := &tar.Header{
+			Name: filepath.Join(baseDir, fmt.Sprintf("%s.log", id)),
+			Mode: int64(os.ModePerm),
+			Size: int64(b.Len()),
+		}
+
+		if err := tw.WriteHeader(hdr); err != nil {
+			fatal(b, err)
+		}
+
+		if _, err := tw.Write(b.Bytes()); err != nil {
+			fatal(b, err)
+		}
+
+		if err := tw.Close(); err != nil {
+			fatal(b, err)
+		}
+	}()
+
+	output(b, stdout, "Starting Launcher Diagnostics\n")
+	output(b, stdout, "ID: %s\n", id)
+	user, err := user.Current()
+	if err != nil {
+		fatal(b, err)
+	}
+	output(b, stdout, "CurrentUser: %s uid: %s\n", user.Username, user.Uid)
+	v := version.Version()
+	jsonVersion, err := json.Marshal(&v)
+	if err != nil {
+		fatal(b, err)
+	}
+	output(b, stdout, "%v\n", string(jsonVersion))
+
+	logger := log.NewLogfmtLogger(b)
+	err = reportGRPCNetwork(
+		logger,
+		serverURL,
+		insecureTLS,
+		insecureGRPC,
+		enrollSecret,
+		certPins,
+		rootPool,
+	)
+	output(b, stdout, "GRPC Connection ...%v\n", err == nil)
+	if err != nil {
+		output(b, fileOnly, "reportGRPCNetwork error: %s", err)
+	}
+
+	err = reportOsqueryProcessInfo(logger, flareSocketPath, b)
+	if err != nil {
+		output(b, fileOnly, "reportOsqueryProcessInfo error: %s", err)
+	}
+	output(b, stdout, "Osqueryi ProcessInfo ...%v\n", err == nil)
+
+	err = reportNotaryPing(logger)
+	if err != nil {
+		output(b, fileOnly, "reportNotaryPing error: %s", err)
+	}
+	output(b, stdout, "Osqueryi Ping Notary ...%v\n", err == nil)
+
+	return nil
+}
+
+type outputDestination int
+
+const (
+	fileOnly outputDestination = iota
+	stdout
+)
+
+func fatal(w io.Writer, err error) {
+	output(w, stdout, "error: %s\n", err)
+	os.Exit(1)
+}
+
+func output(w io.Writer, printTo outputDestination, f string, a ...interface{}) error {
+	if printTo == stdout {
+		fmt.Printf(f, a...)
+	}
+
+	_, err := fmt.Fprintf(w, f, a...)
+	return err
+}
+
+// starts an osqueryd runtime, and then connects an osquery client and runs queries to check and log process info.
```

Comment:
Building this, and running `./launcher flare`, my logs show:
```
reportOsqueryProcessInfo error: creating osquery instance for process info query: starting instance: osqueryd not supplied and not found: exec: "osqueryd": executable file not found in $PATH
```

Reply:
do you not have an osqueryd in your path?

Reply:
I will try to first find a binary is `/usr/local/kolide/bin/osqueryd`, since that's the default in our packages. 
Its not full proof with custom launcher builds though, so we can parametrize it.

Reply:
Correct. I don't have an osquery in my path.